crates/flox/src/commands/channel.rs

```diff
@@ -70,40 +83,99 @@ impl Search {
             })
             .sorted_by(|a, b| Ord::cmp(a, b));
 
-        if self.json {
-            let mut map = serde_json::Map::new();
-            for (channel, entry) in channels {
-                map.insert(
-                    entry.from.id.to_string(),
-                    json!({
-                        "type": channel.to_string(),
-                        "url": entry.to.to_string()
-                    }),
-                );
-            }
+        // Create `registry` parameter for `pkgdb`
+        let mut priority: Vec<String> = Vec::new();
+        let mut inputs = serde_json::Map::new();
+        for (_, entry) in channels {
+            // TODO: handle `subtrees` and `stabilities`
+            priority.push(entry.from.id.to_string());
+            inputs.insert(
+                entry.from.id.to_string(),
+                json!({"from": entry.from.to_string()}),
+            );
+        }
+        let registry = json!({
+            "priority": priority,
+            "inputs": inputs,
+            "default": json!({
+                "subtrees": null, // TODO
+                "stabilities": null,  // TODO
+            })
+        });
+
+        // Create `query` parameter for `pkgdb`
+        let query = match self.search_term {
+            Some(search_term) => {
+                let mut query = serde_json::Map::new();
+                query.insert("match".to_string(), json!(search_term));
+                query
+            },
+            None => serde_json::Map::new(),
+        };
 
-            println!(
-                "{}",
-                serde_json::to_string_pretty(&serde_json::Value::Object(map))?
-            )
-        } else {
-            let width = channels
-                .clone()
-                .map(|(_, entry)| entry.from.id.len())
-                .reduce(|acc, e| acc.max(e))
-                .unwrap_or(8);
+        let params = json!({
+            "registry": registry,
+            "query": query,
+            // These defaults are set by `pkgdb` but are shown here
+            // for visibility.
+            // "semver": { "preferPreReleases": false }
+            // "systems": ["x86_64-linux", ...]
+        });
+
+        let pkgdb_bin: String = match env::var("PKGDB_BIN") {
+            Ok(val) => val,
+            Err(_) => "pkgdb".to_string(),
+        };
```

Comment:
We should instead make this a static, similar to https://github.com/flox/flox/blob/04600ae3c512db6268d637f1f854372812516047/crates/flox-rust-sdk/src/environment.rs#L6

Reply:
I tweaked it so that a static `PKGDB_FALLBACK` is stashed at build time, but `PKGDB` may still be used at runtime to explicitly override it.

Reply:
Suggestion:
```
diff --git a/crates/flox/src/commands/channel.rs b/crates/flox/src/commands/channel.rs
index 133d183..9f77840 100644
--- a/crates/flox/src/commands/channel.rs
+++ b/crates/flox/src/commands/channel.rs
@@ -11,6 +11,7 @@ use flox_rust_sdk::nix::flake_ref::git_service::{GitServiceAttributes, GitServic
 use flox_rust_sdk::nix::flake_ref::FlakeRef;
 use flox_rust_sdk::nix::RunJson;
 use itertools::Itertools;
+use once_cell::sync::Lazy;
 use regex::Regex;
 use serde_json::json;

@@ -29,13 +30,10 @@ enum ChannelType {
     Flox,
 }

-// Set at compile time, used as a fallback when `PKGDB` env variable is unset.
-pub static PKGDB_FALLBACK: &str = env!("PKGDB_BIN");
-
-// This is the `PKGDB` path that we actually use.
-// This is set once and prefers the `PKGDB` env variable, but will use
-// the fallback if it is unset.
-static mut PKGDB_BIN: Option<String> = None;
+// If the environment variable `PKGDB` is set then use it, otherwise
+// fall back to the default `pkgdb` binary provided at build time.
+pub static PKGDB_BIN: Lazy<String> =
+    Lazy::new(|| env::var("PKGDB").unwrap_or(env!("PKGDB_BIN").to_string()));

 /// Search packages in subscribed channels
 #[derive(Bpaf, Clone)]
@@ -130,22 +128,6 @@ impl Search {
             // "systems": ["x86_64-linux", ...]
         });

-        // If the environment variable `PKGDB` is set then use it, otherwise
-        // fall back to the default `pkgdb` binary provided at build time.
-        let pkgdb_bin: String;
-        unsafe {
-            match PKGDB_BIN {
-                Some(_) => {},
-                None => {
-                    PKGDB_BIN = Some(match env::var("PKGDB") {
-                        Ok(val) => val,
-                        Err(_) => PKGDB_FALLBACK.to_string(),
-                    });
-                },
-            }
-            pkgdb_bin = PKGDB_BIN.clone().unwrap();
-        }
-
         // TODO: I have no idea how to check verbosity here.
         //// By default we run with `--quiet`, but if verbose is set then
         //// stream stderr.
@@ -156,7 +138,7 @@ impl Search {
         //maybe_quiet.push("--quiet".to_string());
         let maybe_quiet: Vec<String> = vec!["--quiet".to_string()];

-        let output = Command::new(pkgdb_bin)
+        let output = Command::new(PKGDB_BIN.as_str())
             .arg("search")
             .args(maybe_quiet)
             .arg(params.to_string())
```